workflowRDF/fairworkflow.py

```diff
@@ -165,39 +185,37 @@ def get_result(self):
         return self.result
 
     def get_rdf(self):
-        return self.rdf
-
-    def __str__(self):
-        return self.get_rdf().serialize(format='turtle').decode("utf-8")
 
+        # Autogenerate rdf metadata for this step
+        rdf = rdflib.Graph()
 
-def FairStep(fairworkflow):
-    def fair_wrapper(func):
-        def metadata_wrapper(*args, **kwargs):
+        rdf.add((self.THISSTEP['step'], RDF.type, PPLAN.Step))
+        rdf.add((self.THISSTEP['step'], RDF.type, BPMN.scriptTask))
 
-            # Autogenerate rdf metadata for this step
-            rdf = rdflib.Graph()
+        for var, arg in zip(self.func.__code__.co_varnames, self.args):
+            rdf.add((self.THISSTEP[var], RDF.type, PPLAN.Variable))
+            rdf.add((self.THISSTEP['step'], PPLAN.hasInputVar, self.THISSTEP[var]))
 
-            this_step = PLEX[func.__name__]
+        # Grab entire function's source code for step 'description'
+        func_src = inspect.getsource(self.func)
+        rdf.add((self.THISSTEP['step'], DC.description, rdflib.Literal(func_src)))
 
-            rdf.add((this_step, RDF.type, PPLAN.Step))
-            rdf.add((this_step, RDF.type, BPMN.scriptTask))
-            rdf.add((this_step, PPLAN.isStepOfPlan, fairworkflow.this_workflow))
+        return rdf
+ 
+    def nanopublish(self):
+        nanopuburl = nanopublish(assertionrdf=self.get_rdf(), uri=self.np_uri)
+        self.STEP = rdflib.Namespace(nanopuburl + '#')
```

Comment:
Why does this information need to be saved?

Reply:
Relates to the above point. STEP stores the nanopublished URI so that any workflow RDF being generated can use the correct URI.